Makefile

```diff
@@ -1,12 +1,19 @@
+all: thrift proto
+
+.PHONY: thrift proto
+
 # LightStep-specific: rebuilds the LightStep thrift protocol files.  Assumes
 # the command is run within the LightStep development environment (i.e. the
 # LIGHTSTEP_HOME environment variable is set).
-.PHONY: thrift proto
 thrift:
 	thrift --gen go:package_prefix='github.com/lightstep/lightstep-tracer-go/',thrift_import='github.com/lightstep/lightstep-tracer-go/thrift_0_9_2/lib/go/thrift' -out . $(LIGHTSTEP_HOME)/go/src/crouton/crouton.thrift
 	rm -rf lightstep_thrift/reporting_service-remote
 
 proto:
-	@git submodule update --init
-	cd lightstep-tracer-common && protoc --go_out=plugins=grpc:. --go_out=../collectorpb/ collector.proto
-	@rm lightstep-tracer-common/collector.pb.go
+	@if [ ! -r lightstep-tracer-common/collector.proto ]; then \
+	  echo "Must run 'git submodule update --init' before generating protobuf stubs"; \
+	  false; \
+	fi
+	docker run -v $(shell pwd)/lightstep-tracer-common:/input:ro -v $(shell pwd)/collectorpb:/output \
```

Comment:
Actually not that bad since I sort of copied the thrift version of this... I got more confused by running
```sh
git submodule update --init
```
on every make invocation, since that was changing git state (and reverting the submodule to a previous committed version).